js/gdm/loginDialog.js

```diff
@@ -778,11 +908,13 @@ const LoginDialog = new Lang.Class({
     cancel: function() {
         if (this._verifyingUser)
             this._userVerifier.cancel();
-        else
-            this._reset();
+        this._reset();
     },
 
     _showPrompt: function(forSecret) {
+        if (this._holdForAnswer)
+            throw new Error("Previous _showPrompt not yet finished");
```

Comment:
No, this is an assertion that should never happen. I couldn't find any equivalent of assert() in Gjs, and Philip Chimento suggested throwing an exception instead. I added it for debugging and decided not to remove it because I found it very useful. If this isn't idiomatic, it should be removed.